e2e/test/scenarios/sharing/public-question.cy.spec.js

```diff
@@ -140,6 +136,62 @@ describe("scenarios > public > question", () => {
       });
     }),
   );
+
+  it("should be able to view public questions with snippets", () => {
+    openNativeEditor();
+
+    // Create a snippet
+    cy.icon("snippet").click();
+    cy.findByTestId("sidebar-content").findByText("Create a snippet").click();
+
+    modal().within(() => {
+      cy.findByLabelText("Enter some SQL here so you can reuse it later").type(
+        `'test'`,
+      );
+      cy.findByLabelText("Give your snippet a name").type("string 'test'");
+      cy.findByText("Save").click();
+    });
+
+    cy.get("@editor").type(`{moveToStart}select `);
+
+    saveQuestion("test question", { wrapId: true });
+    cy.get("@questionId").then(id => {
+      cy.request("POST", `/api/card/${id}/public_link`).then(
```

Comment:
we could use `createPublicQuestionLink` which is already imported in this file. This applies to another test too.

Reply:
Done [0d03b3d](https://github.com/metabase/metabase/pull/37405/commits/0d03b3d156cb3369e0d3f53a7f1221e00bc1e7c8)